### PR TITLE
Us40 export reports to csv

### PR DESCRIPTION
**Pull request by msolecka-kainos**
Add creating csv for excel from actual displayed report.
Add reportCsvDataService to create file content.
Add tests to service and controller.

### CODE REVIEW

File: src/main/resources/assets/js/test/report/tdprReportCsvDataServiceTest.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        var reports = [
+            {
+                initHours : 20.5,
+                fullHours : 20,
+                sumOfHours : 20.4,
+                person : {
+                    lastName : "FirstLastName",
+                    firstName : "FirstFirstName"
+                }
+
```
Comment (marcint-kainos) [inline src/main/resources/assets/js/test/report/tdprReportCsvDataServiceTest.js]: Unnecessary line

File: src/main/resources/assets/js/application/report/services/tdprReportCsvDataService.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+define(['angular', 'application/report/tdprReportModule'], function (angular, tdprReportModule) {
+    tdprReportModule.service('tdprReportCsvDataService', ['FileSaver', function (FileSaver) {
+
+        var baseLink = "text/csv;charset=utf-8,";
```
Comment (rlewan) [inline src/main/resources/assets/js/application/report/services/tdprReportCsvDataService.js]: `contentType` would be a more suitable variable name.

File: src/main/resources/assets/js/application/report/services/tdprReportCsvDataService.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            var title = "Report-" + startDate + "-to-" + endDate + ".csv";
+            return title;
+        };
+
+        var createEmptyLine = function() {
+            return separator + separator + separator + endOfLine;
+        };
+
+        var createInfo = function(startDate, endDate) {
+            var info = addQuotes("Start Date:") + separator + addQuotes(startDate) + separator + addQuotes("End Date:") + separator + addQuotes(endDate) + endOfLine;
```
Comment (rlewan) [inline src/main/resources/assets/js/application/report/services/tdprReportCsvDataService.js]: Perhaps [`_.join`](https://lodash.com/docs#join) can be used here?

Comment (rlewan) [inline src/main/resources/assets/js/application/report/services/tdprReportCsvDataService.js]: And in other places where report rows are concatenated.

File: src/main/resources/assets/js/application/report/services/tdprReportCsvDataService.js (hunk 2 above)
Comment (radoslawlandowskikainos) [reply src/main/resources/assets/js/application/report/services/tdprReportCsvDataService.js]: done (y)

File: src/main/resources/assets/js/application/report/services/tdprReportCsvDataService.js (hunk 3 above)
Comment (radoslawlandowskikainos) [reply src/main/resources/assets/js/application/report/services/tdprReportCsvDataService.js]: done (y)

**Pull request merged**